src/frontend/frontend/settings.js

```diff
@@ -31,13 +31,18 @@ angular.module('frontend-app')
           tab_width  : 2,
           use_space : true,
           force_narrow : false,
-          theme : "dark"
+          theme : "dark",
+          offline_mode : 1 /** Valid options: 0 - disabled, 1 - enabled, 2 - forced */
         };
         self.notify = {};
+        self.needToLoad = true;
         var nKey = 0;
 
         function notifyChanges () {
           _.forEach(self.notify, function (fn) {fn();});
+          // Notify of offline/online changes 
+          if (self.settings.offline_mode === "2")
```

Comment:
Should pull offline_mode out into a locally stored cookie, and expose that as a service to avoid these weird service dependencies..